pyproject.toml

```diff
@@ -1,11 +1,143 @@
 [build-system]
-# Minimum requirements for the build system to execute.
-requires = ["setuptools>=61.0.0", "setuptools_scm>=7.0.1"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.0.0", "hatch-vcs"]
+build-backend = "hatchling.build"
 
-[tool.setuptools_scm]
-write_to = "src/pyhf/_version.py"
-local_scheme = "no-local-version"
+[project]
+name = "pyhf"
+dynamic = ["version"]
+description = "pure-Python HistFactory implementation with tensors and autodiff"
+readme = "README.rst"
+license = { text = "Apache-2.0" }  # SPDX short identifier
+requires-python = ">=3.8"
+authors = [
+    { name = "Lukas Heinrich", email = "lukas.heinrich@cern.ch" },
+    { name = "Matthew Feickert", email = "matthew.feickert@cern.ch" },
+    { name = "Giordon Stark", email = "gstark@cern.ch" },
+]
+maintainers = [ {name = "The Scikit-HEP admins", email = "scikit-hep-admins@googlegroups.com"} ]
+keywords = [
+    "fitting",
+    "jax",
+    "numpy",
+    "physics",
+    "pytorch",
+    "scipy",
+    "tensorflow",
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "click>=8.0.0",  # for console scripts
+    "importlib_resources>=1.4.0; python_version < '3.9'",  # for resources in schema
+    "jsonpatch>=1.15",
+    "jsonschema>=4.15.0",  # for utils
+    "pyyaml>=5.1",  # for parsing CLI equal-delimited options
+    "scipy>=1.5.1",  # requires numpy, which is required by pyhf and tensorflow
+    "tqdm>=4.56.0",  # for readxml
+    "numpy",  # compatible versions controlled through scipy
+]
+
+[project.scripts]
+pyhf = "pyhf.cli:cli"
+
+[project.urls]
+Documentation = "https://pyhf.readthedocs.io/"
+Homepage = "https://github.com/scikit-hep/pyhf"
```

Comment:
I probably added this, but I don't think we need it?

Reply:
I think this is just the `url` field from `setup.cfg` getting remapped

https://github.com/scikit-hep/pyhf/blob/3eef1fff2e6d4ccf474bbe51e2211cf690752b82/setup.cfg#L6

I'm fine to keep it for the time being.

Reply:
I mean to say that it's now duplicated here, in `"Source Code"` too. Your call!

Reply:
Yeah we've already been duplicating it:

https://github.com/scikit-hep/pyhf/blob/3eef1fff2e6d4ccf474bbe51e2211cf690752b82/setup.cfg#L14